pdr_backend/analytics/predictoor_dashboard/dash_components/callbacks.py

```diff
@@ -138,14 +139,42 @@ def update_predictoors_table_on_search(search_value, predictoors_data):
         [
             Input("search-input-Feeds", "value"),
             Input("feeds-data", "data"),
+            Input("toggle-switch-predictoor-feeds", "value"),
+            Input("predictoors_table", "selected_rows"),
         ],
+        State("predictoors_table", "data"),
+        State("data-folder", "data"),
     )
-    def update_feeds_table_on_search(search_value, feeds_data):
-        if not search_value:
-            return feeds_data
+    def update_feeds_table_on_search(
+        search_value,
+        feeds_data,
+        predictoor_feeds_only,
+        predictoors_table_selected_rows,
+        predictoors_data,
+        lake_dir,
+    ):
+        # Extract selected predictoor addresses
+        predictoors_addrs = [
```

Comment:
I would suggest adding a function outside of the callbacks, with the purpose of extracting **selected** rows from the predictoors/feeds table. Even if it is a one-liner, it seems specific enough to add, since we repeat it in multple places.